tests/unit/models/primo.js

```diff
@@ -713,6 +785,112 @@ describe("Primo Model >", function() {
+    });
+
+    it("should enhance print records when configuration property is undefined or null", function() {
+      expect(primo.showPrintRecords()).toEqual(true);
+    });
+
+    it("should enhance print records when configuration property is true", function() {
+      browzine.printRecordsIntegrationEnabled = true;
+      expect(primo.showPrintRecords()).toEqual(true);
+    });
+
+    it("should not enhance print records when configuration property is false", function() {
+      browzine.printRecordsIntegrationEnabled = false;
+      expect(primo.showPrintRecords()).toEqual(false);
+    });
+  });
+
+  describe("primo model isFiltered method >", function() {
+    beforeEach(function() {
+      delete browzine.printRecordsIntegrationEnabled;
+    });
+
+    afterEach(function() {
+      delete browzine.printRecordsIntegrationEnabled;
+    });
+
+    it("should not filter electronic records when print records configuration property is undefined or null", function() {
+      var scope = {
+        result: {
+          delivery: {
+            deliveryCategory: ["Alma-E"]
+          }
+        }
+      };
+
+      expect(primo.isFiltered(scope)).toEqual(false);
+    });
+
+    it("should not filter electronic records when print records configuration property is true", function() {
+      var scope = {
+        result: {
+          delivery: {
+            deliveryCategory: ["Alma-E"]
+          }
+        }
+      };
+      browzine.printRecordsIntegrationEnabled = true;
+      expect(primo.isFiltered(scope)).toEqual(false);
+    });
+
+    it("should not filter electronic records when print records configuration property is false", function() {
+      var scope = {
+        result: {
+          delivery: {
+            deliveryCategory: ["Alma-E"]
+          }
+        }
+      };
+      browzine.printRecordsIntegrationEnabled = false;
+      expect(primo.isFiltered(scope)).toEqual(false);
+    });
+
+    it("should not filter print records when print records configuration property is undefined or null", function() {
+      var scope = {
+        result: {
+          delivery: {
+            deliveryCategory: ["Alma-P"]
+          }
+        }
+      };
+
+      expect(primo.isFiltered(scope)).toEqual(false);
+    });
+
+    it("should not filter print records when print records configuration property is true", function() {
+      var scope = {
+        result: {
+          delivery: {
+            deliveryCategory: ["Alma-P"]
+          }
+        }
+      };
+      browzine.printRecordsIntegrationEnabled = true;
+      expect(primo.isFiltered(scope)).toEqual(false);
+    });
+
+    it("should filter print records when print records configuration property is false", function() {
+      var scope = {
+        result: {
+          delivery: {
+            deliveryCategory: ["Alma-P"]
+          }
+        }
+      };
+      browzine.printRecordsIntegrationEnabled = false;
+      expect(primo.isFiltered(scope)).toEqual(true);
+    });
+  });
+
```

Comment:
Same here - maybe a test for primo to not display journal cover images and BZ Web links in the actual DOM element we construct when those settings are set to false?

Reply:
These Unit tests can only test layout we produce: e.g. `directToPDFTemplate` and `browzineWebLinkTemplate`. We don't have one for Journal Covers because we're able to inject into the exiting Primo template: https://github.com/thirdiron/browzine-discovery-service-adapters/blob/develop/src/primo/browzine-primo-adapter.js#L330

But definitely get it though, these "DOM" tests would be at the Acceptance Test level since they require the whole UI down to run. So instead of specific functions, we'd need to setup a top-level scenario for each of those flag tests. Taking a look...

Reply:
Yep. Checkout this Acceptance Test:
https://github.com/thirdiron/browzine-discovery-service-adapters/blob/develop/tests/acceptance/browzine-primo-adapter-test.js#L203

Something like that. Working through these new ones now. Thanks!

Reply:
Thanks @tiandavis ! Hopefully it's not _too_ painful, but it will probably help us avoid any nasty regressions in the future 😊

Reply:
[Acceptance Tests](https://github.com/thirdiron/browzine-discovery-service-adapters/pull/53/commits/5eb15828dc0a3efcfca5c2b0caae630a18c3df68) or the customization properties are added. Should be all set now @karlbecker!